go-update/src/main/java/org/reactome/release/goupdate/GoTermInstanceModifier.java

```diff
@@ -362,10 +379,27 @@ private void pointAllReferrersToOtherInstance(GKInstance replacementGOTerm) thro
 					// remove *this* goInstance from the referrer
 					referrerAttributeValues = referrerAttributeValues.parallelStream().filter(v -> !v.getDBID().equals(this.goInstance.getDBID())).collect(Collectors.toList());
 					// add the replacement to the referrer
-					referrerAttributeValues.add(replacementGOTerm);
-					referrer.setAttributeValue(attributeName, referrerAttributeValues);
+					if (attribute.isMultiple())
+					{
+						referrerAttributeValues.add(replacementGOTerm);
+						referrer.setAttributeValue(attributeName, referrerAttributeValues);
+					}
+					else
+					{
+						referrer.setAttributeValue(attributeName, replacementGOTerm);
+					}
+					// The old Perl code would update referrers' displayNames if they were PhysicalEntities or CatalystActivities.
+					if (referrer.getSchemClass().isa(ReactomeJavaConstants.PhysicalEntity) || referrer.getSchemClass().isa(ReactomeJavaConstants.CatalystActivity))
+					{
+						String newReferrerDisplayName = InstanceDisplayNameGenerator.generateDisplayName(referrer);
```

Comment:
Is InstanceDisplayNameGenerator in release-common-lib? This would be useful for Orthoinference

Reply:
`InstanceDisplayNameGenerator` is in reactome-base, I believe. It originates somewhere in the CuratorTool.

Reply:
You can see the code here: https://github.com/reactome/CuratorTool/blob/master/src/org/gk/model/InstanceDisplayNameGenerator.java